tests/test.py

```diff
@@ -0,0 +1,101 @@
+import unittest
+from homework import Rectangle
+import math
+
+
+class TestInvalidRectangleProperties(unittest.TestCase):
+
+    def setUp(self):
+        self.rect = Rectangle
+        self.invalid_args = ((0, 0), (-1, -1), (-1, 0),
+                             (0, -1), (1, 0), (0, 1),
+                             (1, -1), (-1, 1))
+
+    def tearDown(self) -> None:
+        del self.rect
+
+    @unittest.skip("This version of Rectangle class doesn't support assessment of values")
+    def test_invalid_params(self):
+        for args in self.invalid_args:
+            with self.subTest(args=args):
```

Comment:
every test should raise ValueError here. it's no reason to create subtests here

Reply:
Well, firstly I tried to do this test without subTest, but after the first iteration if test fail it stops cycle, thus it checks only the very first param in invalid params. The self.subTest helps to avoid this situation.